### PR TITLE
Added flexible options for the occupancy ratio calculation for nomina…

### PR DESCRIPTION
**Pull request by efranzis**
…tor and denominator

Total area included into fExcludeAreaExcludedJets condition

### CODE REVIEW

**Comment by efranzis** [comment]
Hi @fkrizek,

please have a look if this version fulfills all your needs. It unfortunately opened a new PR instead of adding to the old one #8012.

Kind Regards,
Eliane

File: PWGJE/EMCALJetTasks/AliAnalysisTaskRhoSparse.cxx [hunk 1]
```diff
@@ -131,57 +134,91 @@ Bool_t AliAnalysisTaskRhoSparse::Run()
...
+    // but to the factor with which this rho is scaled down
+    if (fExcludeAreaExcludedJets==0 && jet->GetNumberOfTracks()>0)
+    {
+      TotaljetAreaPhys+=jet->Area();
+      // Total area of all found jets ghost+phyical jets. This is a proxy
+      // for the available detector area in which the rho could have been calculated
+      TotalAreaCovered+=jet->Area();
+    }
+    // Exclude leading background jets (could be signal)
+    if (iJets == maxJetIds[0] || iJets == maxJetIds[1])
```
Comment (fkrizek) [inline PWGJE/EMCALJetTasks/AliAnalysisTaskRhoSparse.cxx]: Can these lines 161 + 162 be moved before the line 149?

**Comment by fkrizek** [comment]
Hi @efranzis  and  @mfasDa 

i still see that the condition for exclusion of   two leading jets comes only after    summing up jet areas.
I have made note in the code what i think needs to be changed.

Thanks
Filip

File: PWGJE/EMCALJetTasks/AliAnalysisTaskRhoSparse.cxx (hunk 1 above)
Comment (efranzis) [reply PWGJE/EMCALJetTasks/AliAnalysisTaskRhoSparse.cxx]: Hi @fkrizek,

please look at L191. If you set fExcludeAreaExcludedJets==1 which I recommended, then the area is summed after the jets are excluded and L153-159 is ignored. I think this is what you want

Comment (fkrizek) [reply PWGJE/EMCALJetTasks/AliAnalysisTaskRhoSparse.cxx]: Hi @efranzis.

yes, you are right!   I guess you can commit the code then.
Cheers
Filip

**Pull request closed without merging**